BrightID/src/components/Apps/model.ts

```diff
@@ -254,14 +255,24 @@ const linkAppId = async (appId: string, appUserId: string) => {
       const res = await api.linkAppId(sig, appUserId);
     } catch (e) {
       console.log(e);
-      Alert.alert(i18next.t('apps.alert.title.linkingFailed'), `${e.message}`, [
-        {
-          text: i18next.t('common.alert.dismiss'),
-          style: 'cancel',
-          onPress: () => null,
-        },
-      ]);
-      return;
+      if (e instanceof BrightidError && e.errorNum === DUPLICATE_UID_ERROR) {
```

Comment:
I think we should improve how we manage DUPLICATE_UID_ERROR on clients here.

we can not simply ignore this error because it is not happening only when users are trying to link same appId (appUserId). Maybe they forgot they linked before and are trying again.
a better approach seems to be saving the appId that user linked and spent blind sig with in the sig object on client side and when users try to re-use a blind sig, give them an error depending on the appId they are trying to link is old id or a new one.

Reply:
another case that we should handle is when an app has multiple verifications and when user try to link for the first time, he has only one of those verifications, so client ignores linking those verifications. users should be able to try same deep link once more in future to link those new verifications.